### PR TITLE
Feature/600 flyby ekf

### PR DESCRIPTION
**Pull request by thibaudteil**
* **Tickets addressed:** bsk-600
* **Review:** By commit  
* **Merge strategy:** Merge (no squash) 

## Description
Add implementation of a linear filter for orbit determination on a rectilinear trajectory. The filter contains two operational modes, one where velocity is estimated, another where the velocity is set as a constant parameter

## Verification
Unit test included

## Documentation
rst documentation included

## Future work
none

### CODE REVIEW

File: src/fswAlgorithms/opticalNavigation/linearODeKF/linearODeKF.cpp [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+/*! Set a constant velocity vector that will not be estimated. Assert that velocity is not part of the state
+    @return double muCentral
+    */
+void LinearODeKF::setConstantVelocity(const Eigen::Vector3d &velocity){
+    assert(this->state.hasVelocity() == false);
+    this->constantVelocity = velocity;
+}
+
+/*! Get the constant velocity vector
+    @return double muCentral
```
Comment (patkenneally) [inline src/fswAlgorithms/opticalNavigation/linearODeKF/linearODeKF.cpp]: :small_red_triangle_down: Old comment?

File: src/fswAlgorithms/opticalNavigation/linearODeKF/linearODeKF.cpp [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        Eigen::MatrixXd dynMatrix = this->dynamics.computeDynamicsMatrix(t, state);
+        XDot.attachStm(dynMatrix*stm);
+
+        return XDot;
+    };
+    this->dynamics.setDynamics(twoBodyDynamics);
+}
+
+/*! Read the message containing the measurement data.
+ * It updates class variables relating to measurement data including validity and time tags.
```
Comment (patkenneally) [inline src/fswAlgorithms/opticalNavigation/linearODeKF/linearODeKF.cpp]: :small_red_triangle_down: Out of date comment.

**Review by patkenneally — APPROVED** [review]

**Comment by patkenneally** [comment]
There's a lot of copy and paste stuff that has been carried forward from the project's past. Like many of the comments in test scripts. Don't need to action anything. We need to collectively start to remove a lot of that material.

File: src/fswAlgorithms/opticalNavigation/linearODeKF/linearODeKF.cpp (hunk 1 above)
Comment (thibaudteil) [reply src/fswAlgorithms/opticalNavigation/linearODeKF/linearODeKF.cpp]: fixed

File: src/fswAlgorithms/opticalNavigation/linearODeKF/linearODeKF.cpp (hunk 2 above)
Comment (thibaudteil) [reply src/fswAlgorithms/opticalNavigation/linearODeKF/linearODeKF.cpp]: fixed

**Comment by thibaudteil** [comment]
ignoring the random unit test error in the linux build since it's a known issue

**Pull request merged**